cromwell-drs-localizer/src/main/scala/drs/localizer/DrsLocalizerMain.scala

```diff
@@ -59,27 +61,54 @@ class DrsLocalizerMain(drsUrl: String,
     }
   }
 
-  def resolveAndDownloadWithRetries(retries: Int, downloaderFactory: DownloaderFactory,
+  def resolveAndDownloadWithRetries(downloadRetries: Int,
+                                    checksumRetries: Int,
+                                    downloaderFactory: DownloaderFactory,
                                     backoff: Option[CloudNioBackoff],
-                                    attempt: Int = 0): IO[DownloadResult] = {
+                                    downloadAttempt: Int = 0,
+                                    checksumAttempt: Int = 0): IO[DownloadResult] = {
 
-    def maybeResolveAndDownloadWithRetry(t: Throwable): IO[DownloadResult] = {
-      if (attempt < retries) {
+    def maybeRetryForChecksumFailure(t: Throwable): IO[DownloadResult] = {
+      if (checksumAttempt < checksumRetries) {
         backoff foreach { b => Thread.sleep(b.backoffMillis) }
-        logger.warn(s"Attempting retry $attempt of $retries retries to download $drsUrl", t)
-        resolveAndDownloadWithRetries(retries, downloaderFactory, backoff map { _.next }, attempt + 1)
+        logger.warn(s"Attempting retry $checksumAttempt of $checksumRetries checksum retries to download $drsUrl", t)
+        resolveAndDownloadWithRetries(downloadRetries, checksumRetries, downloaderFactory, backoff map { _.next }, downloadAttempt, checksumAttempt + 1)
```

Comment:
I wonder if we should reset `downloadAttempt` to `0` here. We know we've successfully downloaded the file once, so we can be pretty confident we can do it again. It would be a shame to get that far and then fail the job an a transient error.